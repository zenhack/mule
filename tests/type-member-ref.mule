#XFAIL: type member
let x = { type t = int }
in (4 : x.t)
