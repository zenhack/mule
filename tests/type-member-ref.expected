
Desugared:

	(letrec (types ())
	 (values
	  ((x (letrec (types (((t (rec t int))))) (values ()) (.type= t {} t)))))
	 (: (Integer 4) (. x t)))

inferred type:

	int

Runtime term:

	(LetRec (((0 (Record ()))) (Const (Integer 4))))

Evaluated:

	(Const (Integer 4))
